### PR TITLE
feat: Rust semver-checks workflow against base branch

### PR DESCRIPTION
**Pull request by aborgna-q**
Test PR: https://github.com/CQCL/hugr/pull/1598

Limitations:
- Comments won't be updated when the PR title is changed (added/removed a `!` flag).
  It will only change when a new commit is pushed.

### CODE REVIEW

**Comment by aborgna-q** [comment]
It should be working now!

File: .github/workflows/rs-semver-checks.yml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  semver-checks:
+    name: Rust semver-checks 🦀
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_PAT }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          path: PR_BRANCH
+      - name: Checkout baseline
```
[the remark this answers is not in the document]
Comment (aborgna-q) [reply .github/workflows/rs-semver-checks.yml:36]: That's the default behaviour, and it's actually what `semver-checks-action` does.
The problem is that the output is not really useful for the PR author, as once a breaking change gets merged will reappear as noise on every subsequent PR.
We'll still get the aggregated comparison against the last release on the `release-plz` PR, so no need to repeat it here.

`s-c-action` plans to add the baseline checks at some point, but it requires some changes to its definition first.
See the discussion in https://github.com/obi1kenobi/cargo-semver-checks-action/issues/64
For now the alternative is to manually run `semver-checks`, as we do here.

[the remark this answers is not in the document]
Comment (ss2165) [reply .github/workflows/rs-semver-checks.yml:36]: Got it, I agree that release-plz is handling my problem

**Review by ss2165 — APPROVED** [review]
🚀

**Pull request merged**